data-structures-stacks/max-value.js

```diff
@@ -0,0 +1,7 @@
+export default function maxValue(stack) {
+  const allValues = [];
+  while (stack.peek() !== undefined) {
+    allValues.push(stack.pop());
+  }
+  return Math.max(...allValues);
```

Comment:
This is clever but is more work than necessary. See if you can do it without creating a copy of the values.